src/category_theory/graded_object.lean

```diff
@@ -0,0 +1,197 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.shift
+import category_theory.limits.shapes.zero
+
+/-!
+# The category of graded objects
+
+For any type `β`, a `β`-graded object over some category `C` is just
+a function `β → C` into the objects of `C`.
+We define the category structure on these.
+
+We describe the `comap` functors obtained by precomposing with functions `β → γ`.
+
+As a consequence a fixed element (e.g. `1`) in an additive group `β` provides a shift
+functor on `β`-graded objects
+
+When `C` has coproducts we construct the `total` functor `graded_object β C ⥤ C`,
+show that it is faithful, and deduce that when `C` is concrete so is `graded_object β C`.
+-/
+
+open category_theory.limits
+
+namespace category_theory
+
+universes w v u
+
+/-- A type synonym for `β → C`, used for `β`-graded objects in a category `C`. -/
+def graded_object (β : Type w) (C : Type u) : Type (max w u) := β → C
+
+-- Satisfying the inhabited linter...
+instance inhabited_graded_object (β : Type w) (C : Type u) [inhabited C] :
```

Comment:
Won't the linter complain that the instances don't have docstrings?

Reply:
No, the linter isn't concerned about instances, for better or worse. @fpvandoorn, have you considered adding this requirement?